cg/meta/upload/scout/uploadscoutapi.py

```diff
@@ -248,99 +248,111 @@ def upload_rna_coverage_bigwig_to_scout(self, case_id: str, dry_run: bool) -> No
         """Upload rna_coverage_bigwig file for a case to Scout."""
 
         status_db: Store = self.status_db
-        rna_case = status_db.get_case_by_internal_id(internal_id=case_id)
         rna_dna_sample_case_map: Dict[str, Dict[str, list]] = self.create_rna_dna_sample_case_map(
-            rna_case=rna_case
+            rna_case=status_db.get_case_by_internal_id(internal_id=case_id)
         )
         for rna_sample_id in rna_dna_sample_case_map:
             rna_coverage_bigwig: Optional[File] = self.get_rna_coverage_bigwig(
                 case_id=case_id, sample_id=rna_sample_id
             )
-
             if rna_coverage_bigwig is None:
                 raise FileNotFoundError(
                     f"No RNA coverage bigwig file was found in housekeeper for {rna_sample_id}"
                 )
-
             LOG.info(f"RNA coverage bigwig file {rna_coverage_bigwig.path} found")
-            dna_sample_id: str
-            dna_cases: List[str]
-            dna_sample_id, dna_cases = rna_dna_sample_case_map[rna_sample_id].popitem()
-            for dna_case_id in dna_cases:
-                if self.status_db.get_case_by_internal_id(internal_id=dna_case_id).is_uploaded:
-                    LOG.info(
-                        f"Uploading RNA coverage bigwig file for {dna_sample_id} in case {dna_case_id} in scout"
-                    )
-
-                    if dry_run:
-                        continue
-
-                    self.scout_api.upload_rna_coverage_bigwig(
-                        file_path=rna_coverage_bigwig.full_path,
-                        case_id=dna_case_id,
-                        customer_sample_id=dna_sample_id,
-                    )
-                    LOG.info(
-                        f"Uploaded RNA coverage bigwig file for {dna_sample_id} in case {dna_case_id}"
-                    )
-                else:
-                    LOG.warning(
-                        f"Upload of RNA coverage bigwig file for {dna_sample_id} in case {dna_case_id} skipped - "
-                        f"case has not finished uploading"
-                    )
+            self._upload_splice_junctions_bed_or_coverage_for_rna_sample(
+                file=rna_coverage_bigwig,
+                rna_sample_id=rna_sample_id,
+                rna_dna_sample_case_map=rna_dna_sample_case_map,
+                dry_run=dry_run,
+                is_splice_junctions=False,
+            )
+        LOG.info(
+            self.summary_for_rna_case_splice_junctions_or_coverage_upload(
+                rna_dna_sample_case_map=rna_dna_sample_case_map, is_splice_junctions=False
+            )
+        )
         LOG.info("Upload RNA coverage bigwig file finished!")
 
     def upload_splice_junctions_bed_to_scout(self, dry_run: bool, case_id: str) -> None:
```

Comment:
I feel like `upload_rna_coverage_bigwig_to_scout` and `upload_splice_junctions_bed_to_scout` are too similar after the changes you made. I am wondering if they could be merged and add conditionals inside to check for either of the two conditions, so basically merging with the function `_upload_splice_junctions_bed_or_coverage_for_rna_sample`.

Reply:
Yeah, I think that was actually the base problem. Fixed it now - can you check if this is preferrable? @diitaz93

Reply:
It sort of violates the single responsibility principle, though. A bit unsure of what is preferred here.

Reply:
Maybe we can leave it for another PR